### ISSUE TITLE
Does "form control label" still need to look at `aria-labelledby`? (cc0f0a)

### ISSUE THREAD

**Issue by Jym77** [act-rules/act-rules.github.io#1025]
Follow up to https://github.com/act-rules/act-rules.github.io/pull/973#discussion_r347860532, https://github.com/act-rules/act-rules.github.io/pull/973#discussion_r347864920 and #1022.

Since the recent update of the "form control label is descriptive" rule (#986), the rule is considering that "WCAG labels" are for sighted users and applicability has been updated.
However, the rule still targets elements that are referenced by `aria-labelledby` attribute. These elements are not programmatically associated with the form control for sighted users. `label` elements are (clicking on a `label` element jumps focus to the associated form control). But the `aria-labelledby` reference is completely transparent for sighted users. Therefore, there is no strong reason to target them.

---
Using the `aria-labelledby` association can be seen as a good heuristic for figuring out what should be the "WCAG label" for that form control. But using them also runs into a chain of further issues that are tricky to tackle correctly.
1. (completely overlooked by the current rule): `aria-labelledby` can point to several elements. The "label" that should be considered is the full result of this construction, not individual elements.
```html
<span id="billing">Billing</span>
<span id="street">Street</span>
<span id="number">Number</span>
<input type="text" aria-labelledby="billing street number" />
```
we should check whether "Billing Street Number" is descriptive. Currently, the rule checks whether "Billing" is descriptive, whether "Street" is descriptive, and whether "Number" is descriptive" (and fails thrice instead of passing once). With the addition of visual context which is worked on in #973, the rule will check whether "Billing" + context "Street Number" is descriptive, whether "Street" + context "Billing XXX Number" is descriptive, and whether "Number" + context "Billing Street" is descriptive. This is better but not really ideal…

2. We could, typically in the definition suggested in https://github.com/act-rules/act-rules.github.io/pull/973#discussion_r347864920, replaced the single `aria-labelledby` reference by "the result of computing the accessible name when processing the `aria-labelledby` element" (or something similar). However, the computation of accessible name can easily go through non-visible stuff:
```html
<span id="billing" aria-label="foo">Billing</span>
<span id="street">Street</span>
<span id="number">Number</span>
<input type="text" aria-labelledby="billing street number" />
```
The result of the `aria-labelledby` computation is "foo Street Number" which is not descriptive, but given that "foo" is not visible, that is not a problem for the rule which assumes labels are for sighted users and should totally consider "Billing Street Number" in that case.

3. Trying to get into "the visible text content" of these elements or something like that, gets hairy for reasons detailed in https://github.com/act-rules/act-rules.github.io/pull/452#discussion_r344111743

---
## Suggestion 1:
Go for the easy way. Target only `label` elements. That makes sense given that the rule is already restricted to sighted users and sighted users do not see the `aria-labelledby` association.

## Suggestion 2:
Wait for the "visible text content" issue to be resolved in #452 (maybe by https://github.com/act-rules/act-rules.github.io/pull/452#discussion_r345654486) and adapt accordingly.

**Comment by WilcoFiers** [comment]
There's a simpler solution. In the case of multiple labels, treat the other labels as part of the context.

**Comment by Jym77** [comment]
Yes, I've seen that (see point 1 above).
That multiplies the test targets, which is not a problem for the rule but still a bit annoying, and seems a bit of a stretch to consider that something referred by `aria-labelledby` is context for other parts of the label. 

Given that I don't see a strong reason to consider `aria-labelledby` now that the rule is only for sighted users (who do not see the `aria-labelledby` relation), I'm not super happy about that solution. I'll probably do it anyway…
